README.md

```diff
@@ -27,4 +56,134 @@ See
 * [CRUDModel](api/src/main/java/dev/akif/crud/CRUDModel.java)
 * [CRUDEntity](api/src/main/java/dev/akif/crud/CRUDEntity.java)
 
-and their `Simple`, `Simpler` and `Simplest` variants.
+and their Simple, Simpler, and Simplest variants.
+
+## Code Examples
+
+### Writing a controller
+
+The code snippet represents a Java Controller that implements the CRUDController interface.
+
+The CRUDController is a base type that provides endpoints for basic CRUD (Create, Read, Update, Delete) operations. 
+It is the entry point for incoming HTTP requests, and it delegates to a CRUDService for business logic. 
+By extending the CRUDController, developers can quickly set up a working RESTful API for their data models.
+
+```java
+@RestController
+public class MyController implements CRUDController<
+        Long,
+        MyEntity,
+        MyModel,
+        MyDTO,
+        MyCreateModel,
+        MyUpdateModel,
+        MyCreateDTO,
+        MyUpdateDTO,
+        MyMapper,
+        MyDTOMapper,
+        MyService> {
+
+    public MyController(MyService service, MyDTOMapper dtoMapper) {
+        super(service, dtoMapper);
+    }
+}
+
+```
+
+The CRUDController uses generics to ensure that the correct types are used for each layer of the application. The following types are defined as generic parameters:
+
+- `I`: the type of the ID of the entities
+- `E`: the type of the Entity objects
+- `M`: the type of the Model objects
+- `D`: the type of the Data Transfer Object (DTO) objects
+- `CM`: the type of the Create Model objects
+- `UM`: the type of the Update Model objects
+- `CD`: the type of the Create DTO objects
+- `UD`: the type of the Update DTO objects
+- `Mapper`: the type of the mapper class used to map between Entity and Model objects
+- `DTOMappe`r: the type of the mapper class used to map between DTO and Model objects
+- `S`: the type of the CRUDService implementation
+
+By using these generics, the CRUDController ensures that the correct types are used for each layer of the application, and it ensures that everything works together seamlessly.
+
+To use the CRUDController in your application, simply extend it with your specific types and provide implementations for the abstract methods.
+
+```java
+@Service
+public class ExampleService implements CRUDService<
```

Comment:
`CRUDService` is an abstract class so this statement and following example would need to be updated.